SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKDevAuthTokenGenerator.cpp

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "SpatialGDKDevAuthTokenGenerator.h"
+
+#include "Async/Async.h"
+#include "Framework/Notifications/NotificationManager.h"
+
+#include "SpatialCommandUtils.h"
+#include "SpatialGDKEditorSettings.h"
+#include "SpatialGDKSettings.h"
+
+DEFINE_LOG_CATEGORY(LogSpatialGDKDevAuthTokenGenerator);
+
+FSpatialGDKDevAuthTokenGenerator::FSpatialGDKDevAuthTokenGenerator()
+	: bIsGenerating(false)
+{
+}
+
+void FSpatialGDKDevAuthTokenGenerator::DoUpdateSettings(FString DevAuthToken)
+{
+	AsyncTask(ENamedThreads::GameThread, [this, DevAuthToken]()
+		{
+			USpatialGDKEditorSettings* GDKEditorSettings = GetMutableDefault<USpatialGDKEditorSettings>();
+			GDKEditorSettings->DevelopmentAuthenticationToken = DevAuthToken;
+			GDKEditorSettings->SaveConfig();
+			GDKEditorSettings->SetRuntimeDevelopmentAuthenticationToken();
+
+			// Ensure we enable bUseDevelopmentAuthenticationFlow when using cloud deployment flow.
+			USpatialGDKSettings* GDKRuntimeSettings = GetMutableDefault<USpatialGDKSettings>();
+			GDKRuntimeSettings->bUseDevelopmentAuthenticationFlow = true;
+			GDKRuntimeSettings->DevelopmentAuthenticationToken = DevAuthToken;
+
+			this->ShowTaskEndedNotification(TEXT("Developer Authentication Token Updated"), SNotificationItem::CS_Success);
+		});
+}
+
+void FSpatialGDKDevAuthTokenGenerator::DoGenerateDevAuthToken()
```

Comment:
```suggestion
void FSpatialGDKDevAuthTokenGenerator::GenerateDevAuthToken()
```

I don't think we need the Do in front of it. It already says with it's doing. same for `DoUpdateSettings`

Reply:
Raymon asked me not to use the name GenerateDevAuthToken

Reply:
why?

Reply:
Because we have several places to use this function name, so I prefer to use `GenerateDevAuthTokenInner` or similar  name here to distinguish them.

Reply:
it's now called `DoGenerateDevAuthTokenTasks`. How about just `GenerateDevAuthTokenTasks`?